File: desafio3.py
class Heroi:
    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        if self.tipo == "Mago":
            ataque = "Magia"
            
        elif self.tipo == "Guerreiro":
            ataque = "Espada"
            
        elif self.tipo == "Monge":
            ataque = "Artes Marciais"
            
        elif self.tipo == "Ninja":
            ataque = "Shuriken"
        
        return f"{self.nome} O {self.tipo} atacou usando {ataque}."

def coletar_dados():
    while True:
        nome = input("Digite o nome do herói: ")
        idade = input("Digite a idade do herói: ")
        tipo = input("Digite o tipo do herói (Mago, Guerreiro, Monge, Ninja): ")
        
        heroi = Heroi(nome, idade, tipo)
        print(heroi.atacar())

coletar_dados()

File: desafio3.js
class Classe_Heroi{
    constructor(nome, idade, tipo){
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;
    }
    atacar(){
        if (this.tipo === "Mago") {
            const ataque = "Magia";
            console.log(`O ${this.tipo} atacou usando ${ataque}`);

        } else if (this.tipo === "Guerreiro") {
            const ataque = "Espada";
            console.log(`O ${this.tipo} atacou usando ${ataque}`);

        } else if (this.tipo === "Monge") {
            const ataque = "Artes Marciais";
            console.log(`O ${this.tipo} atacou usando ${ataque}`);
            
        } else if (this.tipo === "Ninja") {
            const ataque = "Shuriken";
            console.log(`O ${this.tipo} atacou usando ${ataque}`);

        } 
        }
    }

const heroi1 = new Classe_Heroi("Merlin", 150, "Mago");
heroi1.atacar();

const heroi2 = new Classe_Heroi("Elend", 25, "Guerreiro");
heroi2.atacar();

const heroi3 = new Classe_Heroi("Sazed", 34, "Monge");
heroi3.atacar();

const heroi4 = new Classe_Heroi("Sasuke", 23, "Ninja");
heroi4.atacar();
